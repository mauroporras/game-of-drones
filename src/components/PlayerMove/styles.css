.PlayerMove {
  border: 0.1rem solid black;
  width: 45%;
}

.PlayerMove__player-name {
  display: block;
  font-size: 1.25rem;
  text-align: center;
}
