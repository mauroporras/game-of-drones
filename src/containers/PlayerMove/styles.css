.PlayerMove {
  border: 0.1rem solid black;
  box-sizing: border-box;
  padding: 1rem;
  width: 45%;
}

.PlayerMove__player-name {
  display: block;
  font-size: 1.25rem;
  margin-bottom: 0.25rem;
  text-align: center;
}

.PlayerMove__instruction {
  display: block;
  color: blue;
  margin-bottom: 1rem;
  text-align: center;
}

.PlayerMove__notification {
  display: block;
  color: green;
  margin-bottom: 1rem;
  text-align: center;
}
