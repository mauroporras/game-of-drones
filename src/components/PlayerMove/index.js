import React from 'react'

import ButtonMove from '../ButtonMove/'

import './styles.css'

class PlayerMove extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      done: false,
      moveId: null,
    }

    this._setMove = this._setMove.bind(this)
  }

  _setMove(id) {
    this.setState({
      done: true,
      moveId: id
    })
  }

  render() {
    const doneElement = (
      <span className='PlayerMove__done'>Done!</span>
    )

    const pickElement = (
      <span className='PlayerMove__pick'>Pick a move</span>
    )

    const buttonsElements = this.props.moves.map(e => (
      <ButtonMove
        key={e.id}
        move={e}
        onClick={this._setMove}
      />
    ))

    return (
      <div className='PlayerMove'>
        <strong className='PlayerMove__player-name'>
          {this.props.player.name}
        </strong>
        <div>
          {this.state.done && doneElement}
          {!this.state.done && pickElement}
          {!this.state.done && buttonsElements}
        </div>
      </div>
    )
  }
}

export default PlayerMove
