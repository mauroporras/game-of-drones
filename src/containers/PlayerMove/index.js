import React from 'react'

import Button from '../../components/Button/'

import './styles.css'

class PlayerMove extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      move: null,
    }

    this._setMove = this._setMove.bind(this)
  }

  _setMove(move) {
    this.setState({
      move
    })
    // Emit player and their move.
    this.props.onPick(this.props.player, move)
  }

  render() {
    const notificationElement = (
      <span className='PlayerMove__notification'>Done!</span>
    )

    const instructionElement = (
      <span className='PlayerMove__instruction'>Pick a move</span>
    )

    const buttonsElements = this.props.moves.map(e => (
      <Button
        key={e.id}
        onClick={() => this._setMove(e)}
        text={e.id}
      />
    ))

    const { move } = this.state

    return (
      <div className='PlayerMove'>
        <strong className='PlayerMove__player-name'>
          {this.props.player.name}
        </strong>
        {move && notificationElement}
        {!move && instructionElement}
        {!move && buttonsElements}
      </div>
    )
  }
}

export default PlayerMove
