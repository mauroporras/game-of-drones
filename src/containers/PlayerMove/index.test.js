import React from 'react'
import renderer from 'react-test-renderer'

import PlayerMove from './index'

test("Component renders the correct text", () => {
  const component = renderer.create(
    <PlayerMove
      moves={[]}
      player={{ name: 'Foo' }}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
