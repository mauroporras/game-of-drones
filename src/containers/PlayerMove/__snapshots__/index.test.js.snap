exports[`test Component renders the correct text 1`] = `
<div
  className="PlayerMove">
  <strong
    className="PlayerMove__player-name">
    Foo
  </strong>
  <span
    className="PlayerMove__instruction">
    Pick a move
  </span>
</div>
`;
